core/src/main/java/com/klaytn/caver/tx/type/AbstractTxType.java

```diff
@@ -84,14 +85,59 @@ public AbstractTxType(BigInteger nonce, BigInteger gasPrice, BigInteger gasLimit
         this.from = from;
         this.to = to;
         this.value = value;
+        this.senderSignatureDataList = new ArrayList();
     }
 
     protected void setSenderSignatureData(KlaySignatureData signatureData) {
-        this.senderSignatureData = signatureData;
+        addSenderSignatureData(signatureData);
     }
 
     public KlaySignatureData getSenderSignatureData() {
-        return senderSignatureData;
+        try {
+            return senderSignatureDataList.get(0);
+        } catch (Exception e){
+            return null;
+        }
+    }
+
+    protected void addSenderSignatureData(KlaySignatureData signatureData) {
+        for (KlaySignatureData senderSignatureData : senderSignatureDataList) {
+            if (senderSignatureData.equals(signatureData)) {
+                return;
+            }
+        }
+        senderSignatureDataList.add(signatureData);
+    }
+
+    public void addSenderSignatureDataList(List<KlaySignatureData> signatureDataList) {
+        for (KlaySignatureData signatureData : signatureDataList) {
+            addSenderSignatureData(signatureData);
+        }
+    }
+
+    protected void addSenderSignatureData(List<RlpType> signatureRlpTypeList) {
+        for (RlpType signature : signatureRlpTypeList) {
+            RlpList vrs = (RlpList) signature;
+            byte[] v = ((RlpString) vrs.getValues().get(0)).getBytes();
```

Comment:
Don't we need to have `try` here?

Reply:
Did you mention that `try` is necessary because of the size of vrs.getValues?
?

Reply:
Yes. Is it true that vrs.getValues() always has three items?

Reply:
Perhaps this code did, but I will insert code to check the size.